include/tvm/tir/transform.h

```diff
@@ -437,10 +437,11 @@ TVM_DLL Pass ConvertBlocksToOpaque();
  *
  *  \endcode
  *
- *
+ * \param is_strict ensure the compacted shape always smaller than the original shape.
+ *   otherwise it allows to grow the shape to match actual accessed buffer regions.
  * \return The pass.
  */
-TVM_DLL Pass CompactBufferAllocation();
+TVM_DLL Pass CompactBufferAllocation(bool is_strict = true);
```

Comment:
Should the `is_strict` argument be a parameter in the `PassContext` instead?

Reply:
Emmmm I am preferring to use it as a parameter since we are planning to use it multiple times under the same context., like
- In phase0: standard compaction same with the mainline
- In following phases : configure it as non-strict to adjust buffer regions after we enlarge or align  loop domains to match certain hardware constraints.

Reply:
That makes sense, and thank you for the clarification!